src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -86,20 +87,32 @@ public class SqlMutationEngine : IMutationEngine
             IOutputType outputType = context.Selection.Field.Type;
             string entityName = outputType.TypeName();
             ObjectType _underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
-
+            bool isPointMutation;
             if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingFieldType.Directives, out string? modelName))
             {
+                isPointMutation = true;
                 entityName = modelName;
             }
+            else
+            {
+                isPointMutation = false;
+                outputType = _underlyingFieldType.Fields[MutationBuilder.ARRAY_INPUT_ARGUMENT_NAME].Type;
+                _underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+                if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingFieldType.Directives, out string? baseModelName))
+                {
+                    entityName = baseModelName;
+                }
+            }
 
             ISqlMetadataProvider sqlMetadataProvider = _sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
             IQueryEngine queryEngine = _queryEngineFactory.GetQueryEngine(sqlMetadataProvider.GetDatabaseType());
 
             Tuple<JsonDocument?, IMetadata?>? result = null;
             EntityActionOperation mutationOperation = MutationBuilder.DetermineMutationOperationTypeBasedOnInputType(graphqlMutationName);
 
+            string inputArgumentName = isPointMutation ? MutationBuilder.ITEM_INPUT_ARGUMENT_NAME : MutationBuilder.ARRAY_INPUT_ARGUMENT_NAME;
             // If authorization fails, an exception will be thrown and request execution halts.
-            AuthorizeMutationFields(context, parameters, entityName, mutationOperation);
+            AuthorizeMutationFields(inputArgumentName, context, IMutationEngine.GetClientRoleFromMiddlewareContext(context), parameters, entityName, mutationOperation);
```

Comment:
There is a function in the same class : `GetRoleOfGraphQLRequest` which extracts the role from the hotchocolate context. It can be re-used here

Reply:
Moved that method to authz resolver.